From a61069f4ea924bbf48a5b63c78d30eae7b4a447a Mon Sep 17 00:00:00 2001
From: Balazs Scheidler <bazsi@balabit.hu>
Date: Sat, 2 Nov 2013 20:52:22 +0100
Subject: [PATCH] dbparser: fix memory leak on reload

Upstream-Status: Backport

At init time a reload was forced, but erroneously a new PatternDB instance
was created and the old one leaked. This is obviously not good for
a number of reasons:

 1) memory leak
 2) db-parser() forgot its correllation state
 3) db-parser() forgot its idea of the current time

Signed-off-by: Balazs Scheidler <bazsi@balabit.hu>
---
 modules/dbparser/dbparser.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/modules/dbparser/dbparser.c b/modules/dbparser/dbparser.c
index ee7d637..f1248b5 100644
--- a/modules/dbparser/dbparser.c
+++ b/modules/dbparser/dbparser.c
@@ -149,7 +149,6 @@ log_db_parser_init(LogPipe *s)
         }
       else if (self->db_file_inode != st.st_ino || self->db_file_mtime != st.st_mtime)
         {
-          self->db = pattern_db_new();
           log_db_parser_reload_database(self);
           self->db_file_inode = st.st_ino;
           self->db_file_mtime = st.st_mtime;
-- 
1.8.4.1

